packages/ref-metrics-esmvaltool/src/cmip_ref_metrics_esmvaltool/metrics/base.py

```diff
@@ -142,7 +142,7 @@ def build_metric_result(
             metadata = yaml.load(metadata_file.read_text(encoding="utf-8"))
             for filename in metadata:
                 caption = metadata[filename].get("caption", "")
-                relative_path = Path(filename).relative_to(result_dir.absolute())
+                relative_path = definition.as_relative_path(filename)
```

Comment:
This didn't work for me when running the diagnostic with https://github.com/Climate-REF/climate-ref/blob/main/docs/how-to-guides/running-metrics-locally.py

Reply:
I think I've resolved this in https://github.com/Climate-REF/climate-ref/pull/250/commits/ab0a27f293a58fdd96bea3e7d2aa015d951bc89d

There were some differing assumptions about relative/absolute paths

Reply:
The creation of the output directory happens in the executors which is why it is needed here when running the metric directly